source/common/modules/markdown-editor/plugins/render-images.ts

```diff
@@ -1,202 +1,227 @@
+      openExternally.classList.add('open-externally-button')
+      openExternally.textContent = 'Open image externally'
+      openExternally.onclick = function (event) {
+        // NOTE: We can only do this because the main process prevents any
+        // navigation, and will open the "URL" using the shell.
+        window.location.assign(actualURLToLoad)
+      }
+
+      const figure = document.createElement('figure')
+      figure.appendChild(img)
+      figure.appendChild(caption)
+      figure.appendChild(size)
+
+      const container = document.createElement('div')
+      container.classList.add('editor-image-container')
+      container.appendChild(figure)
+
+      // Now add a line widget to this line.
+      const textMarker = cm.markText(
+        curFrom,
+        curTo,
+        {
+          'clearOnEnter': true,
+          'replacedWith': container,
+          'handleMouseEvents': false
+        }
+      )
+
+      // Define a quick inline function that takes care of applying a new caption
+      const updateCaptionFunction = function (event: KeyboardEvent|FocusEvent): void {
+        if (event instanceof KeyboardEvent && event.key !== 'Enter') {
+          // If this is a KeyboardEvent, only perform the action on Enter
+          return
+        }
+
+        // Get the correct, current position where the token is now
+        const currentPosition = textMarker.find()
+
+        if (currentPosition === undefined) {
+          return // Could not locate the textmarker
+        }
+
+        event.preventDefault()
+        event.stopPropagation()
+        // Make sure there are no quotes since these will break the image
+        const newCaption = caption.textContent?.replace(/"/g, '') ?? ''
+        // "Why are you setting the caption both as the image description and title?"
+        // Well, since all exports sometimes us this, sometimes the other value.
+        const newImageTag = `![${newCaption}](${url} "${newCaption}")${p4}`
+        // Clear the text marker so that this hook can immediately rerender it.
+        textMarker.clear()
+        // Then replace the range, which will trigger a "change" event which in
+        // turn will rerender this image.
+        cm.replaceRange(newImageTag, currentPosition.from, currentPosition.to)
+      }
+
+      // Should work on these events
+      caption.addEventListener('keydown', updateCaptionFunction)
+      caption.addEventListener('focusout', updateCaptionFunction)
+
+      // Retrieve the size constraints
+      const maxPreviewWidth = Number((cm as any).getOption('zettlr').imagePreviewWidth)
+      const maxPreviewHeight = Number((cm as any).getOption('zettlr').imagePreviewHeight)
+      let width = (!Number.isNaN(maxPreviewWidth)) ? `${maxPreviewWidth}%` : '100%'
+      let height = (!Number.isNaN(maxPreviewHeight) && maxPreviewHeight < 100) ? `${maxPreviewHeight}vh` : ''
+
+      // Use the pandoc attributes to control image size
+      if (widthAttribute !== '') {

```

Comment:
What would need to change if you were to preset those two attributes with `undefined` instead? That might make more semantic sense, and would probably alleviate, e.g., the needs for the `?? ''` operator above?